src/parser.rs

```diff
@@ -3438,173 +3438,199 @@ impl<'a> Parser<'a> {
-            }
-
-            AlterTableOperation::ChangeColumn {
-                old_name,
-                new_name,
-                data_type,
-                options,
-            }
-        } else if self.parse_keyword(Keyword::ALTER) {
-            let _ = self.parse_keyword(Keyword::COLUMN);
-            let column_name = self.parse_identifier()?;
-            let is_postgresql = dialect_of!(self is PostgreSqlDialect);
-
-            let op = if self.parse_keywords(&[Keyword::SET, Keyword::NOT, Keyword::NULL]) {
-                AlterColumnOperation::SetNotNull {}
-            } else if self.parse_keywords(&[Keyword::DROP, Keyword::NOT, Keyword::NULL]) {
-                AlterColumnOperation::DropNotNull {}
-            } else if self.parse_keywords(&[Keyword::SET, Keyword::DEFAULT]) {
-                AlterColumnOperation::SetDefault {
-                    value: self.parse_expr()?,
-                }
-            } else if self.parse_keywords(&[Keyword::DROP, Keyword::DEFAULT]) {
-                AlterColumnOperation::DropDefault {}
-            } else if self.parse_keywords(&[Keyword::SET, Keyword::DATA, Keyword::TYPE])
-                || (is_postgresql && self.parse_keyword(Keyword::TYPE))
-            {
-                let data_type = self.parse_data_type()?;
-                let using = if is_postgresql && self.parse_keyword(Keyword::USING) {
-                    Some(self.parse_expr()?)
+            Keyword::INDEX => {
+                let _ = self.parse_keyword(Keyword::ONLY);
+                let index_name = self.parse_object_name()?;
+                let operation = if self.parse_keyword(Keyword::RENAME) {
+                    if self.parse_keyword(Keyword::TO) {
+                        let index_name = self.parse_object_name()?;
+                        AlterIndexOperation::RenameIndex { index_name }
+                    } else {
+                        return self.expected("TO after RENAME", self.peek_token());
+                    }
                 } else {
-                    None
+                    return self.expected("RENAME after ALTER INDEX", self.peek_token());
                 };
-                AlterColumnOperation::SetDataType { data_type, using }
-            } else {
-                return self.expected(
-                    "SET/DROP NOT NULL, SET DEFAULT, SET DATA TYPE after ALTER COLUMN",
-                    self.peek_token(),
-                );
-            };
-            AlterTableOperation::AlterColumn { column_name, op }
-        } else {
-            return self.expected(
-                "ADD, RENAME, PARTITION or DROP after ALTER TABLE",
-                self.peek_token(),
-            );
-        };
-        Ok(Statement::AlterTable {
-            name: table_name,
-            operation,
-        })
+
+                Ok(Statement::AlterIndex {
+                    name: index_name,
+                    operation,
+                })
+            }
+            _ => unreachable!(),
```

Comment:
How should we handle this unreachable? 🤔 
Is there a way to get fit enum of keywords that i only need (TABLE, INDEX)?

Reply:
I think this is ok and a consequence of using `expect_one_of_keywords`

Perhaps a comment

```suggestion
            // unreachable because expect_one_of_keywords used above
            _ => unreachable!(),
```